packages/client/components/promptResponse/PromptResponseEditor.tsx

```diff
@@ -45,6 +45,14 @@ const PromptResponseEditor = (props: Props) => {
   const {autoFocus: autoFocusProp, content, handleSubmit, readOnly, placeholder} = props
   const [_isEditing, setIsEditing] = useState(false)
   const [autoFocus, setAutoFocus] = useState(autoFocusProp)
+  const [linkMenuProps, setLinkMenuProps] = useState<
+    {text: string; href: string; originCoords: BBox} | undefined
```

Comment:
+1 nit: since LinkMenu & LinkPreview are mutually exclusive, you could express that relation by reusing the same state.